Source/Core/VideoCommon/BPStructs.cpp

```diff
@@ -753,61 +753,808 @@ void LoadBPRegPreprocess(u32 value0)
+    return format_to(ctx.out(),
+                     "Enable: {}\n"
+                     "Logic ops: {}\n"
+                     "Dither: {}\n"
+                     "Color write: {}\n"
+                     "Alpha write: {}\n"
+                     "Dest factor: {}\n"
+                     "Source factor: {}\n"
+                     "Subtract: {}\n"
+                     "Logic mode: {}",
+                     no_yes[mode.blendenable], no_yes[mode.logicopenable], no_yes[mode.dither],
+                     no_yes[mode.colorupdate], no_yes[mode.alphaupdate], dstfactors[mode.dstfactor],
+                     srcfactors[mode.srcfactor], no_yes[mode.subtract], logicmodes[mode.logicmode]);
+  }
+};
+
+template <>
+struct fmt::formatter<ConstantAlpha>
+{
+  constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+  template <typename FormatContext>
+  auto format(const ConstantAlpha& c, FormatContext& ctx)
+  {
+    return format_to(ctx.out(),
+                     "Enable: {}\n"
+                     "Alpha value: {:02x}",
+                     c.enable ? "Yes" : "No", c.alpha);
+  }
+};
+
+template <>
+struct fmt::formatter<PEControl>
+{
+  constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+  template <typename FormatContext>
+  auto format(const PEControl& config, FormatContext& ctx)
+  {
+    const char* pixel_formats[] = {"RGB8_Z24", "RGBA6_Z24", "RGB565_Z16", "Z24",
+                                   "Y8",       "U8",        "V8",         "YUV420"};
+    const char* zformats[] = {
+        "linear",     "compressed (near)",     "compressed (mid)",     "compressed (far)",
+        "inv linear", "compressed (inv near)", "compressed (inv mid)", "compressed (inv far)"};
+    return format_to(ctx.out(),
+                     "EFB pixel format: {}\n"
+                     "Depth format: {}\n"
+                     "Early depth test: {}",
+                     pixel_formats[config.pixel_format], zformats[config.zformat],
+                     config.early_ztest ? "Yes" : "No");
+  }
+};
+
+template <>
+struct fmt::formatter<FieldMask>
+{
+  constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+  template <typename FormatContext>
+  auto format(const FieldMask& mask, FormatContext& ctx)
+  {
+    return format_to(ctx.out(),
+                     "Enable odd field: {}\n"
+                     "Enable even field: {}",
+                     mask.odd ? "Yes" : "No", mask.even ? "Yes" : "No");
+  }
+};
+
+template <>
+struct fmt::formatter<UPE_Copy>
+{
+  constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+  template <typename FormatContext>
+  auto format(const UPE_Copy& copy, FormatContext& ctx)
+  {
+    const char* no_yes[] = {"No", "Yes"};
+    const char* frame_to_field[] = {"Progressive", "Invalid", "Interlaced (even lines)",
+                                    "Interlaced (odd lines)"};
+    std::string clamp;
```

Comment:
```suggestion
    std::string_view clamp;
```

given this is only ever assigned string literals. Ditto for other relevant variables.